src/main/java/org/kainos/ea/api/AuthService.java

```diff
@@ -0,0 +1,61 @@
+package org.kainos.ea.api;
+
+
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.security.Keys;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.kainos.ea.cli.LoginRequest;
+import org.kainos.ea.cli.User;
+import org.kainos.ea.client.FailedLoginException;
+import org.kainos.ea.db.AuthDao;
+
+import java.security.Key;
+import java.util.Date;
+
+public class AuthService {
+    private static final long EXPIRATION_TIME = 86400000; // 24 hours in milliseconds
+
+
+    public AuthService(AuthDao authDao) {
+        this.authDao = authDao;
+
+    }
+
+    private AuthDao authDao;
+
+
+
+
+    public ImmutablePair<User, String> login(LoginRequest login) throws FailedLoginException {
+        try {
+            User user = authDao.validLogin(login);
+            if (user != null) {
+                return new ImmutablePair<>(user, generateToken(user.getEmail()));
+            }else {
+                throw new FailedLoginException();
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
You treat every error as a Failed Login and then throw an Authorised code. Could you improve that and throw an internal server error when it's a technical issue compared to an issue with the data submitted

Reply:
The comment still needs to be addressed.

I would also look for a way to simplify the nested if statements to improve readability.